core-plugins/src/main/java/io/cdap/plugin/batch/action/EmailAction.java

```diff
@@ -94,11 +100,24 @@ public PasswordAuthentication getPasswordAuthentication() {
       javaMailProperties.put("mail.smtp.ssl.enable", true);
     }
     if ("TLS".equalsIgnoreCase(config.protocol)) {
-      javaMailProperties.put("mail.smtp.starttls.enable", "true");
-      javaMailProperties.put("mail.smtp.ssl.trust", config.host);
+      javaMailProperties.put("mail.smtp.starttls.enable", true);
+      if (!userJavaMailProperties.containsKey("mail.smtp.starttls.enable")) {
+        javaMailProperties.put("mail.smtp.starttls.enable", "true");
+      }
+      if (!userJavaMailProperties.containsKey("mail.smtp.ssl.trust")) {
+        javaMailProperties.put("mail.smtp.ssl.trust", config.host);
+      }
       protocolForTransport = "smtp"; // protocol is smtp not tls.
     }
 
+     if (!userJavaMailProperties.containsKey("mail.smtps.ssl.protocols")) {
+       javaMailProperties.put("mail.smtps.ssl.protocols", "TLSv1.2");
```

Comment:
Is there a reason we aren't using TLSv1.3? Is that the one which isn't available without conscrypt?

Reply:
yeah TLSv.12 is the latest version of TLS that works without conscrypt.

Reply:
Gotcha, since the field [accepts a space-delimited list of protocols](https://javaee.github.io/javamail/docs/api/com/sun/mail/smtp/package-summary.html), can we try adding `TLSv1.2 TLSv1.3` as default?